srcjs/notifications.js

```diff
@@ -0,0 +1,160 @@
+exports.notifications = (function() {
+
+  // Milliseconds to fade in or out
+  const fadeDuration = 250;
+
+  function show({ html=null, duration=5000, id=null, closeButton=true,
+                  style = null } = {})
```

Comment:
`class` is a bit misleading too, since the value you pass it won't literally be the CSS class, as we prepend it with `"shiny-notification-"` or whatever... what about "type" or "level" (like a log level)?